File: Epay3Client/src/types/GuestPaymentRequest.ts
import { Invoice, PayerData, SoldToPayment } from './InvoicesSearchRequest';
import {
  AddressData,
  GuestPaymentCardSubmission,
  PayerDetails,
  PaymentDetail,
} from './Payment';
import { ThreeDSCardinalData } from './AccountResponse';

export interface GuestPaymentRequest {
  payer: PayerData;
  payment: PaymentDetail;
  soldTo: SoldToPayment;
  invoices: Invoice[];
  guestUserEmail?: string;
  threeDSAccessToken?: string;
  cardinalData?: ThreeDSCardinalData;
  vRef?: string;
}

export interface GuestPaymentPreAuthorizeCardRequest {
  payerData: PayerDetails;
  paymentCard: GuestPaymentCardSubmission;
  userId?: string;
  addressData?: AddressData;
  vRef?: string;
}

File: Epay3Net/Models/GuestPaymentRequest.cs
using Epay3Service.Models;
using System.ComponentModel.DataAnnotations;

namespace Epay3Net.Models;

public class GuestPaymentRequest
{
    [Required]
    public PayerData Payer { get; set; } = null!;
    [Required]
    public PaymentDetail Payment { get; set; } = null!;

    [Required]
    public SoldToPayment SoldTo { get; set; } = null!;

    [Required]
    public List<PaymentInvoice> Invoices { get; set; } = null!;

    public string? GuestUserEmail { get; set; }

    [MaxLength(2048)]
    public string? ThreeDSAccessToken { get; set; }

    public CardinalData? CardinalData { get; set; }
    
    [MaxLength(256)]
    public string? VRef { get; set; }
}
